kratos/utilities/global_pointer_utilities.h

```diff
@@ -97,8 +97,7 @@ class GlobalPointerUtilities
             const auto it = container.find(id);
             if( it != container.end()) //found locally
             {
-                //if(container[id].FastGetSolutionStepValue(PARTITION_INDEX) == current_rank)
-                if(IteratorIsLocal(it, current_rank))
+                if(!rDataCommunicator.IsDistributed() || ObjectIsLocal(*it, current_rank))
```

Comment:
can t we do the check for is_distributed only.once instead of per every iterstor?

Reply:
mm you are right :S